polaris-react/src/components/Popover/components/PopoverOverlay/tests/PopoverOverlay.test.tsx

```diff
@@ -277,18 +273,116 @@ describe('<PopoverOverlay />', () => {
       </PopoverOverlay>,
     );
 
-    const target = popoverOverlay.find(TextField)!.find('svg')!.domNode;
-    // eslint-disable-next-line import/no-deprecated
-    const clickEventListener = popoverOverlay.find(EventListener, {
-      event: 'click',
-    })!;
+    const target = popoverOverlay.find(TextField)!.find('svg')!.domNode!;
+    const clickEvent = new MouseEvent('click', {bubbles: true});
+
+    target.dispatchEvent(clickEvent);
 
-    clickEventListener!.trigger('handler', {
-      target,
-    });
     expect(spy).not.toHaveBeenCalled();
   });
 
+  it('does not call the onClose callback when another overlay is clicked if preventCloseOnChildOverlayClick is true', () => {
+    const spy = jest.fn();
+
+    const popoverOverlayAndPopover = mountWithApp(
+      <div>
+        <PopoverOverlay
+          active
+          id="PopoverOverlay-1"
+          activator={activator}
+          onClose={spy}
+          preventCloseOnChildOverlayClick
+        >
+          <button id="popover 1 button" />
+        </PopoverOverlay>
+        <Popover
+          active
+          fullWidth
+          activator={<div>Activator</div>}
+          onClose={jest.fn()}
+        >
+          <button id="popover 2 button" />
+        </Popover>
+      </div>,
+    );
+
+    const popover2Button = popoverOverlayAndPopover.find('button', {
+      id: 'popover 2 button',
+    })!.domNode!;
+    const clickEvent = new MouseEvent('click', {bubbles: true});
+
+    popover2Button.dispatchEvent(clickEvent);
+
+    expect(spy).not.toHaveBeenCalled();
+  });
+
+  describe('deleting descendant elements', () => {
+    const DeleteButton = () => {
+      const [show, setShow] = React.useState(true);
+
+      return (
+        <>{show && <button onClick={() => setShow(!show)}>Delete</button>}</>
+      );
+    };
+
+    it('does not call the onClose callback when a descendant that has been removed from the DOM is clicked', () => {
+      const spy = jest.fn();
+
+      const popoverOverlay = mountWithApp(
+        <PopoverOverlay
+          active
+          id="PopoverOverlay-1"
+          activator={activator}
+          onClose={spy}
+        >
+          <DeleteButton />
+        </PopoverOverlay>,
+      );
+
+      const deleteButton = popoverOverlay.find('button')!.domNode!;
+      const clickEvent = new MouseEvent('click', {bubbles: true});
+
+      deleteButton.dispatchEvent(clickEvent);
```

Comment:
```suggestion
      popoverOverlay.find('button')!.trigger('onClick', {bubbles: true});
```
I'm not 100% sure this will work as intended but we typically we do something like this vs dispatching events where we can

Reply:
Interesting. I've seen both in `web`. What would be the reason to prefer `trigger` over `dispatchEvent`?

Reply:
Just more common for click events. Not a blocker but it keeps test suites more consistent. We use dispatch to mock things like window / document listeners more than behavioural interactions

Reply:
I tried to do this and found that invoking `onClick` via `trigger` didn't actually cause the events to bubble the way you expect even when passing `{bubbles:true}`. `trigger` appears to only call the `onClick` prop, so it doesn't test the functionality the way I intend. If you have any other ideas to try let me know, but otherwise I'm going to keep the `dispatchEvent` here because it does mimic the behavior I intend to test.

Reply:
That's strange but all good. Thanks for trying!